### ISSUE TITLE
Issues with XML Validation in ZATCA Portal

### ISSUE THREAD

**Issue by mub009** [SallaApp/ZATCA#42]
Hi,

I am encountering several issues with the web validator for XML in the ZATCA portal. Below are the specific error messages I have received:

Wrong X509IssuerName
HashedXml does not match with QR code hashedXml or the hashedXml is missing
Wrong X509IssuerName
Wrong X509SerialNumber
Wrong Signature Value
TimeStamp does not match with QR code TimeStamp or TimeStamp missing in invoice
These errors seem to depend on the CSR token.

Could you please assist in resolving these issues?

Thank you!

![image](https://github.com/user-attachments/assets/b995cebb-dc71-43da-88ac-4e612ffeae07)

**Comment by bestiony** [comment]
I have the same problem , any luck finding a solution ?

**Comment by bestiony** [comment]
@mub009 have you found the cause of the problem or an alternative package?

**Comment by bestiony** [comment]
@mub009 
I fixed the problem. I was using Laravel Blade to make XML files. 
1. hash problem was caused by the the whitespace in my XML templates . I removed all whitespace and comments 
![image](https://github.com/user-attachments/assets/eeff93a9-2a44-4db9-ac24-5bd809c520e8)

2. the time in QR code does not time in the invoice was caused by the absence of 'Z' after the timestamps. 
* explanation: The Zatca docs state that adding 'Z' is optional but this Sella package always adds it  
![image](https://github.com/user-attachments/assets/ae40da47-eba2-4319-b14c-1f7bd4b35d65)

3. do not use the web XML validator because for some reason it does not work with simulation invoices or maybe it does not work with invoices that were made for the compliance testing phase. I'll keep you updated if that changes
